Source/RESTyard.Extensions.Pagination/Pagination.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace RESTyard.Extensions.Pagination;
+
+/// <summary>
+/// Defines a contract for pagination functionality in queries.
+/// </summary>
+public readonly record struct Pagination(int PageSize, int PageOffset)
+{
+    
+    /// <summary>
+    /// Indicates whether pagination is disabled for the current query.
+    /// </summary>
+    public bool IsDisabled => PageSize == 0 && PageOffset == 0;
```

Comment:
This could even be
```suggestion
    public bool IsDisabled => this == DisabledPagination;
```
but I am also good with this!